pkg/query-service/telemetry/telemetry.go

```diff
@@ -74,28 +75,32 @@ func (a *Telemetry) IsSampled() bool {
 
 }
 
-func (telemetry *Telemetry) CheckSigNozMetricsV2(compositeQuery *model.CompositeMetricQuery) bool {
+func (telemetry *Telemetry) CheckSigNozSignals(postData *v3.QueryRangeParamsV3) (bool, bool) {
+	signozLogsUsed := false
+	signozMetricsUsed := false
 
-	signozMetricsNotFound := false
-
-	if compositeQuery.BuilderQueries != nil && len(compositeQuery.BuilderQueries) > 0 {
-		if !strings.Contains(compositeQuery.BuilderQueries["A"].MetricName, "signoz_") && len(compositeQuery.BuilderQueries["A"].MetricName) > 0 {
-			signozMetricsNotFound = true
+	if postData.CompositeQuery.QueryType == v3.QueryTypeBuilder {
+		for _, query := range postData.CompositeQuery.BuilderQueries {
+			if query.DataSource == v3.DataSourceLogs && len(query.Filters.Items) > 0 {
+				signozLogsUsed = true
+			} else if query.DataSource == v3.DataSourceMetrics && !strings.Contains(query.AggregateAttribute.Key, "signoz_") {
+				signozMetricsUsed = true
+			}
 		}
-	}
-
-	if compositeQuery.PromQueries != nil && len(compositeQuery.PromQueries) > 0 {
-		if !strings.Contains(compositeQuery.PromQueries["A"].Query, "signoz_") && len(compositeQuery.PromQueries["A"].Query) > 0 {
-			signozMetricsNotFound = true
+	} else if postData.CompositeQuery.QueryType == v3.QueryTypePromQL {
+		for _, query := range postData.CompositeQuery.PromQueries {
+			if !strings.Contains(query.Query, "signoz_") && len(query.Query) > 0 {
+				signozMetricsUsed = true
+			}
 		}
-	}
-	if compositeQuery.ClickHouseQueries != nil && len(compositeQuery.ClickHouseQueries) > 0 {
-		if !strings.Contains(compositeQuery.ClickHouseQueries["A"].Query, "signoz_") && len(compositeQuery.ClickHouseQueries["A"].Query) > 0 {
-			signozMetricsNotFound = true
+	} else if postData.CompositeQuery.QueryType == v3.QueryTypeClickHouseSQL {
+		for _, query := range postData.CompositeQuery.ClickHouseQueries {
+			if strings.Contains(query.Query, "signoz_metrics") && len(query.Query) > 0 {
```

Comment:
It would be a good idea to add traces and logs check here too as dashboards can have CH queries on logs and traces as datasource too

Reply:
For traces active user we consider number of services > 0 and for logs we consider if user has used any filter.
Primary objective of this PR is track active user.
Tracking CH query can be added later as it's unlikely that a user writes CH query but don't send traces data or don't use logs filter.

Reply:
We should have different events to track clickhouse queries.